src/components/CommentSummary/index.tsx

```diff
@@ -0,0 +1,90 @@
+import * as React from 'react';
+
+import { gettext } from '../../utils';
+import { Comment, createCommentKey } from '../../reducers/comments';
+import styles from './styles.module.scss';
+
+export type PublicProps = { className?: string; comments: Comment[] };
+
+const CommentSummary = ({ comments, className }: PublicProps) => {
+  const overallComments = [];
+  const fileComments = [];
+
+  for (const comment of comments) {
+    if (comment.filename) {
+      fileComments.push(comment);
+    } else {
+      overallComments.push(comment);
+    }
+  }
+
+  const lines: string[] = [];
+
+  const addComment = (comment: Comment) => {
+    lines.push(comment.content || '');
+    lines.push('');
+  };
+
+  if (overallComments.length) {
+    lines.push(gettext('Overall comment(s):'));
+    lines.push('');
+
+    for (const comment of overallComments) {
+      addComment(comment);
+    }
+  }
+
+  const groups: { [key: string]: Comment[] } = {};
+
+  // Group comments by file / line so they appear under the same header.
+  for (const comment of fileComments) {
+    const key = createCommentKey({
+      fileName: comment.filename,
+      line: comment.lineno,
+    });
+
+    if (!groups[key]) {
+      groups[key] = [];
+    }
+
+    groups[key].push(comment);
+  }
+
+  for (const key of Object.keys(groups)) {
```

Comment:
Would it be worth sorting the file comments first by `filename` and then by `lineno` at some point, to guarantee that they render in sequential order, or is it assumed that the `comments` property will always contain comments in the exact order in which we wish to render them?

Reply:
The comments will be ordered by how they were entered by the reviewer. This is how github does it and I think it's probably what a reviewer would expect. It's probably a good place to start. We can get some feedback from reviewers to see what they think of it after using it.